### PR TITLE
Continue drop q

### PR DESCRIPTION
**Pull request by raphinesse**
Continuation of https://github.com/apache/cordova-android/pull/833

Fixes a few things in your PR and completes the removal except for the cases where we need to switch to `execa` first.

### CODE REVIEW

File: bin/templates/cordova/lib/check_reqs.js [hunk 1]
```diff
@@ -21,6 +21,7 @@
 
 var shelljs = require('shelljs');
 var child_process = require('child_process');
+var Q = require('q');
```
Comment (erisu) [inline bin/templates/cordova/lib/check_reqs.js:24]: This was added back?

File: bin/templates/cordova/lib/check_reqs.js [hunk 2]
```diff
@@ -192,9 +193,7 @@ module.exports.check_java = function () {
             }
         }
     }).then(function () {
-        return new Promise(resolve => {
-            resolve(child_process.execSync('javac -version'));
-        })
+        return Q.denodeify(child_process.exec)('javac -version')
```
Comment (erisu) [inline bin/templates/cordova/lib/check_reqs.js:196]: And here?

**Review by erisu — COMMENTED** [review]
I saw `q` added back in one file. Was this a mistake?

File: bin/templates/cordova/lib/check_reqs.js (hunk 1 above)
Comment (raphinesse) [reply bin/templates/cordova/lib/check_reqs.js:24]: No mistake. Those are the instances removed by the execa PR. The idea was to merge this into your Q PR, then merge the execa PR, rebase your Q PR and merge it. Then all of Q should be gone without us having to create any intermediary workarounds for the spawn stuff

**Pull request merged**